nova/virt/vmwareapi/vm_util.py

```diff
@@ -2053,3 +2056,40 @@ def rename_vm(session, vm_ref, instance):
 def is_vim_instance(o, vim_type_name):
     return isinstance(o, sudsobject.Factory.subclass(vim_type_name,
                                                      sudsobject.Object))
+
+
+def service_locator_name_password(client_factory, username, password):
+    o = client_factory.create('ns0:ServiceLocatorNamePassword')
+    o.username = username
+    o.password = password
+    return o
+
+
+def _get_sha1_ssl_thumbprint(url):
+    """Returns the sha-1 thumbprint of the ssl cert of the vcenter
+    """
+    parsed_url = six.moves.urllib.parse.urlparse(url)
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    sock.settimeout(1)
+    wrappedSocket = ssl.wrap_socket(sock)
+    thumb = None
+    wrappedSocket.connect((parsed_url.hostname, parsed_url.port or 443))
+
+    der_cert_bin = wrappedSocket.getpeercert(True)
+
+    thumb = hashlib.sha1(der_cert_bin).hexdigest().upper()
+    t = iter(thumb)
+    return ':'.join(a + b for a, b in zip(t, t))
+
+
+def service_locator(client_factory, url, vcenter_instance_uuid, credential,
+                    ssl_thumbprint=None):
+    if not ssl_thumbprint:
+        ssl_thumbprint = _get_sha1_ssl_thumbprint(url)
+
+    sl = client_factory.create('ns0:ServiceLocator')
+    sl.url = url
+    sl.instanceUuid = vcenter_instance_uuid
+    sl.credential = credential
```

Comment:
I'm unsure what I can pass here. Can we document it? Does it take the output of `service_locator_name_password()`?

Reply:
> In general, I'm wondering about function naming. Looking at the file, some are `get_` some are `create_` and some are just like the new ones (contain just the name of the thing they create). Not sure what's the right choice here, but just naming it like the thing feels like the worst choice to me. It's shorter in the calling code though ...

I'll go then with the `create_` prefix.
Get sounds to me more like something requiring an api-call